src/java/org/apache/cassandra/db/Keyspace.java

```diff
@@ -299,15 +301,34 @@ public boolean snapshotExists(String snapshotName)
     /**
      * Clear all the snapshots for a given keyspace.
      *
-     * @param snapshotName the user supplied snapshot name. It empty or null,
+     * @param snapshotName the user supplied snapshot name. If empty or null,
      *                     all the snapshots will be cleaned
+     * @param keyspace keyspace to remove snapshots for
      */
     public static void clearSnapshot(String snapshotName, String keyspace)
     {
+        Set<TableSnapshot> snapshotsToClear = new SnapshotLoader().loadSnapshots(keyspace)
+                                                                  .stream()
+                                                                  .filter(ts -> {
+                                                                      if (snapshotName == null || snapshotName.isEmpty())
```

Comment:
I simplified this filtering a bit [here](https://github.com/instaclustr/cassandra/pull/47/files#diff-9bf2c26bc294ef9085e16bf287490223665eaa2eb8ec24bcf5bd8653c713644bR4213) by encapsulating the filtering logic on [TableSnapshot.shouldClearSnapshot](https://github.com/instaclustr/cassandra/pull/47/files#diff-7d6d1bafcad95c5715c91c9065a4a8c58c3d5c98d0699d9c913717f5c0086bb7R328). Let me know what do you think.